src/ORM/ManyManyThroughList.php

```diff
@@ -217,6 +217,9 @@ public function add($item, $extraFields = [])
         $record = $hasManyList->createDataObject($extraFields ?: []);
         $record->$localKey = $itemID;
         $hasManyList->add($record);
+
+        // Link the join object to the $item, as if it were queried from within this list
+        $item->setJoin($record, $this->manipulator->getJoinAlias());
```

Comment:
Righto, good spot. I think this can be simpler - the setJoin() is only needed if $item instanceof DataObject. I'll correct.

I should probably also add a test for this feature before we merge it.